android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -257,6 +261,69 @@ public void onBandwidthSample(int elapsedMs, long bytes, long bitrate) {
     }
 
     // Internal methods
+
+    /**
+     * Toggling the visibility of the player control view 
+     */
+    private void togglePlayerControlVisibility() {
+        reLayout(playerControlView);
+        if(playerControlView.isVisible()) {
+            playerControlView.hide();
+        } else {
+            playerControlView.show();
+        }
+    }
+
+    /**
+     * Initializing Player control
+     */
+    private void initializePlayerControl() {
+        if(playerControlView == null) {
+            playerControlView = new PlayerControlView(getContext());
+        }
+        LayoutParams layoutParams = new LayoutParams(
+                LayoutParams.MATCH_PARENT,
+                LayoutParams.MATCH_PARENT);
+        playerControlView.setLayoutParams(layoutParams);
+        addView(playerControlView, 1, layoutParams);
+
+        //Setting the player for the playerControlView
+        playerControlView.setPlayer(player);
+        playerControlView.show();
+        playPauseControlContainer = playerControlView.findViewById(R.id.exo_play_pause_container);
+
+        //Invoking onClick event for exoplayerView
+        exoPlayerView.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                togglePlayerControlVisibility();
+            }
+        });
+
+        //Invoking onPlayerStateChanged event for Player
+        eventListener = new Player.EventListener() {
+            @Override
+            public void onPlayerStateChanged(boolean playWhenReady, int playbackState) {
+                reLayout(playPauseControlContainer);
+                player.removeListener(eventListener);
```

Comment:
I'm not sure why you're doing this here, can you explain what's going on. Once you remove the listener, it won't ever fire this event again. Is this because if you reLayout once it works after that? It would be good to have a comment saying this if that's the case.

Reply:
> I'm not sure why you're doing this here, can you explain what's going on. Once you remove the listener, it won't ever fire this event again. Is this because if you reLayout once it works after that? It would be good to have a comment saying this if that's the case.

@cobarx 
Yes, Everything will work fine once after the reLayout for the playerViewControl is done. Will add the comment for the same.

Will check the below test harness and let you know the behaviour immediately.
https://github.com/cobarx/react-native-video-test

Reply:
I have fixed the issue observed in the test harness.

Changed the execution order of initializePlayerControl method in order to align with the player state. Since the createView is executed before the "setControls" props method gets called. so the values of the playerControl are not aligned with the player state values.